libtenzir/builtins/aspects/partitions.cpp

```diff
@@ -76,29 +61,32 @@ class plugin final : public virtual aspect_plugin {
       ctrl.abort(std::move(error));
       co_return;
     }
-    auto builder = table_slice_builder{partition_type()};
+    auto builder = series_builder{};
     using namespace tenzir::si_literals;
     constexpr auto max_rows = size_t{8_Ki};
     for (auto i = 0u; i < synopses.size(); ++i) {
       auto& synopsis = synopses[i];
-      if (not(builder.add(fmt::to_string(synopsis.uuid))
-              && builder.add(uint64_t{synopsis.synopsis->memusage()})
-              && builder.add(synopsis.synopsis->min_import_time)
-              && builder.add(synopsis.synopsis->max_import_time)
-              && builder.add(synopsis.synopsis->version)
-              && builder.add(synopsis.synopsis->schema.name())
-              && builder.add(synopsis.synopsis->schema.make_fingerprint()))) {
-        diagnostic::error("failed to add partition entry")
-          .emit(ctrl.diagnostics());
-        co_return;
-      }
+      auto event = builder.record();
+      event.field("uuid").data(fmt::to_string(synopsis.uuid));
+      event.field("memusage").data(synopsis.synopsis->memusage());
+      event.field("events").data(synopsis.synopsis->events);
+      event.field("min_import_time").data(synopsis.synopsis->min_import_time);
+      event.field("max_import_time").data(synopsis.synopsis->max_import_time);
+      event.field("version").data(synopsis.synopsis->version);
+      event.field("schema").data(synopsis.synopsis->schema.name());
+      event.field("schema_id")
+        .data(synopsis.synopsis->schema.make_fingerprint());
       if ((i + 1) % max_rows == 0) {
-        co_yield builder.finish();
-        builder = std::exchange(builder, table_slice_builder{partition_type()});
+        for (auto result : builder.finish_as_table_slice("tenzir.partition")) {
+          co_yield std::move(result);
+        }
```

Comment:
Nitpick: `for (auto& ...)` is one less copy, and same below (although it's cheap anyway).

Reply:
Exactly, that's what we want to do. But if we leave it as-is, then we copy them first and _then_ move them out, right?